src/coom.jl

```diff
@@ -33,36 +33,41 @@ julia> using TextAnalysis, DataStructures
 ```
 """
 function coo_matrix(::Type{T},
-                    doc::Vector{<:AbstractString},
-                    vocab::OrderedDict{<:AbstractString, Int},
-                    window::Int,
-                    normalize::Bool=true) where T<:AbstractFloat
+    doc::Vector{<:AbstractString},
+    vocab::OrderedDict{<:AbstractString,
+        Int},
+    window::Int,
+    normalize::Bool=true,
+    mode::Symbol=:default) where {T<:AbstractFloat}
+    # Initializations
     n = length(vocab)
     m = length(doc)
     coom = spzeros(T, n, n)
     # Count co-occurrences
     for (i, token) in enumerate(doc)
-        row = get(vocab, token, nothing)
-        isnothing(row) && continue
-
-        @inbounds for j in max(1, i - window):min(m, i + window)
-            i == j && continue
-
+        inner_range = if mode == :directional
+            i:min(m, i + window)
+        else
+            max(1, i - window):min(m, i + window)
+        end
+        # looking forward
+        @inbounds for j in inner_range
             wtoken = doc[j]
-            col = get(vocab, wtoken, nothing)
-            isnothing(col) && continue
-
             nm = T(ifelse(normalize, abs(i - j), 1))
-            coom[row, col] += one(T) / nm
-            coom[col, row] = coom[row, col]
+            row = get(vocab, token, nothing)
```

Comment:
`get(vocab, token, nothing)` is called multiple times for the same `token` value in the internal loop. Performance issue.

Reply:
Thanks! I moved it outside the inner loop, a few lines above.